src/Features/CSharp/Portable/Diagnostics/Analyzers/FileBasedPrograms/ExternalHelpers.cs

```diff
@@ -0,0 +1,22 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#if !NET
+using Roslyn.Utilities;
+
+namespace Microsoft.DotNet.FileBasedPrograms;
+
+/// <summary>Provides implementations of certain methods to the FileBasedPrograms source package.</summary>
+internal partial class ExternalHelpers
+{
+    public static partial int CombineHashCodes(int value1, int value2)
```

Comment:
why do we need this?

Reply:
but it's just combining hash codes. :)  why does it need roslyn's version of that?  why can't it just do it itself.  In thisc ase, this is just saving them writing:

```
            return unchecked((currentKey * (int)0xA5555529) + newKey);
```

note: i don't want anyone taking a dep on our actual hash code (which might change at any time).

Reply:
The usage of the hash code is ephemeral, there is no expectation for it to return same results from one run of the analyzer to the next, or from one version of the analyzer implementation to the next.

I can take your feedback over to the source package, it seems reasonable enough, and then we can delete this stub in a follow-up. Thanks